### ISSUE TITLE
Accessibility: [Screen Reader]:  The state of the button (selected or not) is not programmatically associated with the response buttons.   

### ISSUE THREAD

**Issue by ericmaino** [microsoft/BotFramework-WebChat#3083]
![image](https://user-images.githubusercontent.com/1005609/78628200-5a870d00-7848-11ea-8dd6-7d9d81990cf5.png)

**Actual**
A screen reader user does not know which response button (Yes or No) is selected. He has to come out of the response button group and read the next line of text to get that information
 
**Expected**
The state of the button should be programmatically associated using aria-checked or aria-pressed. 
 
**WCAG Violation**
Name, Role Value: 4.1.2
https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html

**Comment by corinagum** [comment]
These are accessibility issues of v3 of Web Chat, which has been deprecated and we no longer support unless there are security risks. For accessible Web Chat, please [migrate to v4](https://github.com/Microsoft/BotFramework-WebChat#migrating-from-web-chat-v3-to-v4).

**Comment by compulim** [comment]
This is a bug on Adaptive Cards.

The content author is creating those buttons using hero card, which is rendered through Adaptive Cards.

@corinagum can you find the AC bug and associate this with it?

**Comment by compulim** [comment]
Although the button looks like a fork in an intersection (if "yes", go to step 1, if "no" go to step 3), the actual design of the button was a single action. These actions are designed to be okay to trigger/click multiple times. It's not a one-off. Thus, the button is not like a radio button or toggle button.

I think the content author should use suggested actions instead. Suggested actions is a feature designed for one-off use. That means, it will be working like a fork in an intersection, the user can only choose A or B, and not going back.

Thus, I think this is misuse of the feature, causing A11y issues.

**Comment by rp-dev** [comment]
This still repros in customer environment after migrating to v4. 

I understand @compulim's response above and however moving to suggested actions would be a change that needs to be taken with health bot team which I am following up on separately. 

Note that in the health bot, no text entry is allowed, when user selects Yes, there is no bubble from user showing yes, the button is simply shown as selected for the visual user. For screen readers going back to the area with yes/no should read which button is selected which it is not doing right now.

This was tested on Win10, Chrome, Jaws and customer deployed health bot.

**Comment by compulim** [comment]
Could you use suggested actions instead of Adaptive Cards? Suggested actions are designed as an one-off decision. If the design of the suggested actions are not great, can you file an enhancement issue?

We also implemented "disable Adaptive Cards button" here, https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/05.custom-components/l.disable-adaptive-cards.

However, the sample can only be used to disable buttons, but not setting `aria-selected` because buttons in AC card are not designed as toggle button and it's outside of our scope to change the design.

**Comment by compulim** [comment]
# Problem statement

Currently, we have carousel (default) and stacked. But it is not enough. People move to Hero Card because they prefer inline buttons that are one-off and disable on submit.

HealthBot also use Hero Card as the solution for QnA messages.

# Work needed

## Design a new "inline" layout for suggested actions

This will replace the need of Adaptive Cards for simpler question-and-answer messages.

Adaptive Cards are designed to be: dynamic and allow multiple submission. This design don't work for QnA, which is static and one-off. And it is causing issues with accessibility, because, we should not disable AC after it is submitted. But on accessibility side, it is confusing not to disable the AC buttons.

Another accessibility requirement: inline buttons can be "re-read" from the screen reader. The end-user can go back and find out what answer they chose.

### Alternatives

Facebook Messenger use "persistent menu" and the UI looks like our suggested actions. The usage is different. In Facebook approach, the buttons are for "switching context", and not "answering questions and continuing the flow".

By switching context, I mean, for example, these are the buttons: "where is my order", "contact support", "previous orders". These buttons are for initiation of largely different dialogs.

## Work with HealthBot team to move from Hero Card to QnA

- Understand what influenced HealthBot team to use Hero Card as QnA
- Also, understand if QnA maker is using Hero Card

**Comment by compulim** [comment]
I am building a prototype for inline suggested actions. The "inline flow" layout would looks like:

https://user-images.githubusercontent.com/1622400/103744735-1da49e00-4fb3-11eb-8ce4-448521aaceff.mp4

You can find the [branch here](https://github.com/microsoft/BotFramework-WebChat/compare/master...compulim:feat-inline-suggested-actions?expand=1).

Few notes on the prototype:

- Need to continue to implement/investigate
  - Command `emptycard` is not working because it does not have a body
  - Need to study the impact on accessibility
    - Need to study the alt text "suggested actions container: empty"
    - Need to find out how access key will work in inline layout
  - Need to highlight selected action
    - Need to remember temporarily (use `markActivity`) or permanently (add it to `channelData` of the next response and do a lookahead)
    - Need to implement ARIA for the selected action
- Decisions so far
  - It seems "inline carousel" does not make much sense, so we didn't enabled it
  - We refactored `<SuggestedActions>` (per-activity) into `<BasicSuggestedActions>` (global)

**Comment by corinagum** [comment]
Adding @Kaiqb to assignees for tracking purposes.
